sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs

```diff
@@ -375,16 +376,18 @@ public async Task EventProcesorCreatesScopeForEventProcessing()
                 new KeyValuePair<string, string>(DiagnosticProperty.EndpointAttribute, fullyQualifiedNamespace)
             };
 
+            var scopes = listener.Scopes.ToList();
```

Comment:
Those these shouldn't be running in parallel, it seems that the common practice for querying `Activity` parts is to make defensive copies and use those since the activity itself is sensitive to the ambient environment.